realm/realm-library/src/androidTest/java/io/realm/RealmJsonTests.java

```diff
@@ -1178,4 +1183,328 @@ public void createObjectFromJson_objectWithPrimaryKeySetValueDirectlyFromStream(
         assertEquals(1, owners.get(1).getId());
         assertEquals("bar", owners.get(1).getName());
     }
+
+    private void createNullablePrimaryKeyObjectFromJson(Realm testRealm, Class <? extends RealmObject> clazz, String jsonString) throws JSONException {
+        testRealm.beginTransaction();
+        testRealm.createObjectFromJson(clazz, new JSONObject(jsonString));
+        testRealm.commitTransaction();
+    }
+
+    private void createOrUpdateNullablePrimaryKeyObjectFromJson(Realm testRealm, Class <? extends RealmObject> clazz, String jsonString) throws JSONException {
+        testRealm.beginTransaction();
+        testRealm.createOrUpdateObjectFromJson(clazz, new JSONObject(jsonString));
+        testRealm.commitTransaction();
+    }
+
+    private void createAndUpdateNullablePrimaryKeyObjectFromJson(Realm testRealm, Class <? extends RealmObject> clazz, String jsonString) throws JSONException {
+        testRealm.beginTransaction();
+        testRealm.createObject(clazz); // name = null, id = 0
+        testRealm.createOrUpdateObjectFromJson(clazz, new JSONObject(jsonString));
+        testRealm.commitTransaction();
+    }
+
+    // Nullable PrimaryKey tests for PrimaryKeyAsString
+    @Test
+    public void createObjectFromJson_stringPrimaryKeyFieldIsNullFromJsonObject() throws JSONException {
```

Comment:
Maybe these could be `createObjectFromJson_stringPrimaryKey_isNull`, createObjectFromJson_stringPrimaryKey_isAbsent`, etc. It would make them easier to scan at a glance.